src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/ProfessionalRefreshOrchestrationHelper.java

```diff
@@ -0,0 +1,337 @@
+                                                      AccessTypesEntity accessTypes) {
+
+        String reference  = userRefreshQueue.getUserId();
+        List<RoleAssignment> usersRoleAssignments;
+        try {
+            usersRoleAssignments = prepareRoleAssignments(userRefreshQueue,accessTypes);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        AssignmentRequest assignmentRequest;
+
+        assignmentRequest = AssignmentRequest.builder()
+                .request(
+                        Request.builder()
+                                .requestType(RequestType.CREATE)
+                                .replaceExisting(true)
+                                .process(PROFESSIONAL_ORGANISATIONAL_ROLE_MAPPING)
+                                .reference(reference)
+                                .assignerId(securityUtils.getUserId())
+                                .clientId(AM_ORG_ROLE_MAPPING_SERVICE)
+                                .correlationId(UUID.randomUUID().toString())
+                                .build())
+                .requestedRoles(usersRoleAssignments)
+                .build();
+
+        return assignmentRequest;
+    }
+
+    private List<RoleAssignment> prepareRoleAssignments(UserRefreshQueueEntity userRefreshQueue,
+                                                        AccessTypesEntity accessTypes) throws JsonProcessingException {
+        Set<RoleAssignment> usersRoleAssignments = new HashSet<>();
+        // Step 2 If user_refresh_queue.deleted != null then halt and return an empty set of role assignments.
+        if (null != userRefreshQueue.getDeleted()) {
+            return usersRoleAssignments.stream().toList();
+        }
+
+        // Step 3 If user_refresh_queue.organisation_status is not an active status, then halt and return an empty set
+        // of role assignments. Note The set of active statuses will have to be provided by PRD team / product owner.
+
+        if (!isOrganisationStatusActive(userRefreshQueue.getOrganisationStatus())) {
+            return usersRoleAssignments.stream().toList();
+        }
+
+        RestructuredAccessTypes prmRestructuredAccessTypes =
+                convertInRestructuredAccessTypes(accessTypes.getAccessTypes());
+
+        Set<OrganisationProfile> organisationProfiles = prmRestructuredAccessTypes.getOrganisationProfiles();
+
+        List<UserAccessType> userAccessTypes = JacksonUtils.convertUserAccessTypes(userRefreshQueue.getAccessTypes());
+
+        //Step 4 Filter PRM access_types to contain only data for the organisation profiles in
+        // user_refresh_queue.organisation_profile_ids.
+
+        Set<OrganisationProfile> filteredOrganisationProfiles =
+                getFilteredOrganisationProfiles(userRefreshQueue, organisationProfiles);
+
+        List<UserAccessType> filteredUserAccessTypes =
```

Comment:
`getFilteredUserAccessTypes(...)`  - I struggle to understand this method. why are we filtering on the user_access_types? why don't we directly filter the PRM stored access_types (using the data from user_access_types) - to retrieve the final filtered version of PRM access types we need?